pandas/core/apply.py

```diff
@@ -71,7 +72,7 @@
     from pandas.core.resample import Resampler
     from pandas.core.window.rolling import BaseWindow
 
-ResType = dict[int, Any]
+_ResType: TypeAlias = dict[int, Any]
```

Comment:
Is this idea of making these lead with a starting `_` is to they are marked as "private"? Should they instead just be moved to `_typing.py` with other defined "private" annotations?

Reply:
We were planning on moving `core` to `_core`, but ran into some bikeshedding issues. I think we should still operate under the assumption we will eventually do that (I believe there is consensus that we _really_ want to do it). Assuming that is done, what will we want to do about private variables like this? I think we should strive for consistency, either all with a leading underscore or none.

Reply:
> IMHO, anything within that file should be private unless it is meant to be imported from somewhere else.

This seems to me to be a very large change. I count 2432 functions (not merely callables) at the top level of modules across pandas that do not start with an `_`. Now some are undoubtedly public, but many are not. On the other hand, there are 438 functions at the top level of a module that do start with an `_`.

Reply:
If we had `_core` instead of `core`, I believe that will be flagged by pyright as a private import.